File: BankE-main/BankE-Backend/BankE.Domain/Entities/Transaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BankE.Domain.Entities
{
    public class Transaction
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int SenderAccountId { get; set; }

        [ForeignKey("SenderAccountId")]
        public virtual Account SenderAccount { get; set; } = null!;

        [Required]
        public int ReceiverAccountId { get; set; }

        [ForeignKey("ReceiverAccountId")]
        public virtual Account ReceiverAccount { get; set; } = null!;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        [MaxLength(200)]
        public string? Description { get; set; }

        [Required]
        [MaxLength(20)]
        public string Status { get; set; } = "Completed";

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: BankE-main/BankE-main/lib/domain/entities/transaction.dart
class TransactionEntity {
  final String id;
  final String type;
  final double amount;
  final String direction;
  final String description;
  final DateTime date;
  final double? balance;

  const TransactionEntity({
    required this.id,
    required this.type,
    required this.amount,
    required this.direction,
    required this.description,
    required this.date,
    this.balance,
  });

  bool get isCredit => direction == 'Credit';
}
